distributor/distributor.go

```diff
@@ -441,61 +434,84 @@ func (d *Distributor) Query(ctx context.Context, from, to model.Time, matchers .
 			return err
 		}
 
-		if len(ingesters) < d.cfg.MinReadSuccesses {
-			return fmt.Errorf("could only find %d ingesters for query. Need at least %d", len(ingesters), d.cfg.MinReadSuccesses)
+		// We need a response from a quorum of ingesters, which is n/2 + 1.
+		minSuccess := (len(ingesters) / 2) + 1
+		maxErrs := len(ingesters) - minSuccess
+		if len(ingesters) < minSuccess {
+			return fmt.Errorf("could only find %d ingesters for query. Need at least %d", len(ingesters), minSuccess)
 		}
 
-		// Fetch samples from multiple ingesters and group them by fingerprint (unsorted
-		// and with overlap).
-		successes := 0
-		var lastErr error
-		for _, ing := range ingesters {
-			client, err := d.getClientFor(ing)
-			if err != nil {
-				return err
-			}
+		req, err := util.ToQueryRequest(from, to, matchers)
+		if err != nil {
+			return err
+		}
 
-			req, err := util.ToQueryRequest(from, to, matchers)
-			if err != nil {
-				return err
-			}
+		// Fetch samples from multiple ingesters
+		var numErrs int32
+		errs := make(chan error)
```

Comment:
The routine below is constructed such that this will only ever receive one error, so the name `errs` is misleading. `tooManyErrs`? `errReceived`? `maxErr`?